compiler/src/dotty/tools/dotc/util/Signatures.scala

```diff
@@ -48,31 +49,42 @@ object Signatures {
    */
   def callInfo(path: List[tpd.Tree], span: Span)(using Context): (Int, Int, List[SingleDenotation]) =
     path match {
+      case (unapply @ UnApply(fun, implicits, patterns)) :: parent :: _ =>
+        // in case of unapply with `using` such as `def unapply(using Context)(o : Any) = ...`
+        val implicitArgs = countParams(unapply.forwardTo)
```

Comment:
Looks like this is the same as:
```suggestion
        val implicitArgs = countParams(fun)
```
which is symmetrical with the Apply case.

Reply:
Ach, you're right! I simplified and inlined it.